svd-parser/src/expand.rs

```diff
@@ -0,0 +1,396 @@
+use anyhow::{anyhow, Result};
+use std::collections::HashMap;
+use std::mem::take;
+use svd_rs::{
+    array::names, cluster, field, peripheral, register, BitRange, Cluster, ClusterInfo, DeriveFrom,
+    Device, EnumeratedValues, Field, Peripheral, Register, RegisterCluster,
+};
+
+#[derive(Clone, Debug, Default)]
+pub struct Index<'a> {
+    pub peripherals: HashMap<String, &'a Peripheral>,
+    pub clusters: HashMap<String, &'a Cluster>,
+    pub registers: HashMap<String, &'a Register>,
+    pub fields: HashMap<String, &'a Field>,
+    pub evs: HashMap<String, &'a EnumeratedValues>,
+}
```

Comment:
Is this intended to be `pub`? I'm guessing not, since no public function exposes it.

infact, all `pubs` in this module could be removed except for on the `expand` method (which should/could be `pub(crate)`